romancal/source_catalog/tests/test_source_catalog.py

```diff
@@ -317,8 +329,8 @@ def test_l3_input_model_unchanged(mosaic_model, tmp_path):
         save_results=False,
     )
 
-    assert_allclose(original_data, mosaic_model.data, atol=5.0e-7)
-    assert_allclose(original_err, mosaic_model.err, atol=5.0e-7)
+    assert_allclose(original_data, mosaic_model.data, atol=5.0e-5)
+    assert_allclose(original_err, mosaic_model.err, atol=5.0e-5)
```

Comment:
Should we worry about the lower tolerance here?  I think the only action we intend to take is to add the background and then subtract it again.  It is possible for that to introduce issues but it seems like they should be very small; I didn't expect this PR to change the accuracy of that?

Reply:
We needed to change the datatype for the `data` array in the `make_test_image()` method to `float32`, which caused larger differences, and, consequently, we had to increase the tolerance by a factor of 100. If we don't set the data array to `float32` then we run into validation issues with saving the mock models.